fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/physical/PhysicalMergeOperator.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.operator.physical;
+
+import com.google.common.collect.Lists;
+import com.starrocks.common.LocalExchangerType;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptExpressionVisitor;
+import com.starrocks.sql.optimizer.RowOutputInfo;
+import com.starrocks.sql.optimizer.operator.ColumnOutputInfo;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+
+import java.util.List;
+
+public class PhysicalMergeOperator extends PhysicalSetOperation {
```

Comment:
> Why not use PhysicalSetOperation? what's the difference?

mostly because planFragment builder's logic for PhysicalSetOperation is not sutiable for skew join. PhysicalSetOperation will always create new fragment and add exchange node between set and it's child.